acceptance_test.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/conduitio-labs/conduit-connector-db2/config"
+	s "github.com/conduitio-labs/conduit-connector-db2/source"
+)
+
+const (
+	queryCreateTestTable       = `CREATE TABLE %s (id int, name VARCHAR(100))`
+	queryDropTestTable         = `DROP TABLE %s`
+	queryDropTestTrackingTable = `DROP TABLE CONDUIT_TRACKING_%s`
+	queryIfExistTable          = `SELECT count(*) AS count FROM  SysCat.Tables WHERE TabName='CONDUIT_TRACKING_%s'`
+)
+
+type driver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+
+	counter int32
+}
+
+// GenerateRecord generates a random sdk.Record.
+func (d *driver) GenerateRecord(t *testing.T, operation sdk.Operation) sdk.Record {
+	atomic.AddInt32(&d.counter, 1)
+
+	return sdk.Record{
+		Position:  nil,
+		Operation: operation,
+		Metadata: map[string]string{
+			config.KeyTable: d.Config.DestinationConfig[config.KeyTable],
+		},
+		Key: sdk.StructuredData{
+			"ID": d.counter,
+		},
+		Payload: sdk.Change{After: sdk.RawData(
+			fmt.Sprintf(
+				`{"ID":%d,"NAME":"%s"}`, d.counter, gofakeit.Name(),
+			),
+		),
+		},
+	}
+}
+
+//nolint:paralleltest // we don't need paralleltest for the Acceptance tests.
+func TestAcceptance(t *testing.T) {
+	cfg := prepareConfig(t)
+
+	sdk.AcceptanceTest(t, &driver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest:        beforeTest(t, cfg),
+			},
+		},
+	})
+}
+
+// beforeTest creates new table before each test.
+func beforeTest(t *testing.T, cfg map[string]string) func(t *testing.T) {
+	return func(t *testing.T) {
+		table := randomIdentifier(t)
+		t.Logf("table under test: %v", table)
+
+		cfg[config.KeyTable] = table
+
+		err := prepareData(t, cfg)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func prepareConfig(t *testing.T) map[string]string {
+	conn := os.Getenv("DB2_CONNECTION")
+	if conn == "" {
+		t.Skip("DB2_CONNECTION env var must be set")
+
+		return nil
+	}
+
+	return map[string]string{
+		config.KeyConnection: conn,
+		config.KeyPrimaryKey: "ID",
+		s.KeyOrderingColumn:  "ID",
+	}
+}
+
+func prepareData(t *testing.T, cfg map[string]string) error {
+	db, err := sql.Open("go_ibm_db", cfg[config.KeyConnection])
+	if err != nil {
+		return err
+	}
+
+	_, err = db.Exec(fmt.Sprintf(queryCreateTestTable, cfg[config.KeyTable]))
+	if err != nil {
+		return err
+	}
+
+	db.Close()
+
+	// drop table
+	t.Cleanup(func() {
+		db, err = sql.Open("go_ibm_db", cfg[config.KeyConnection])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		queryDropTable := fmt.Sprintf(queryDropTestTable, cfg[config.KeyTable])
+
+		_, err = db.Exec(queryDropTable)
+		if err != nil {
+			t.Errorf("drop test table: %v", err)
+		}
+
+		queryDropTrackingTable := fmt.Sprintf(queryDropTestTrackingTable, cfg[config.KeyTable])
+
+		// check if table exist.
+		rows, er := db.Query(fmt.Sprintf(queryIfExistTable, cfg[config.KeyTable]))
```

Comment:
Why not use `err` instead of `er`?

Reply:
Because on this case we shadow variable err

Reply:
yep